tree/dataframe/test/datasource_csv.cxx

```diff
@@ -323,6 +324,107 @@ TEST(RCsvDS, ProgressiveReadingRDFMT)
    EXPECT_EQ(6U, *c2);
 }
 
+TEST(RCsvDS, NaNTypeIndentification)
+{
+   RCsvDS tds(fileName4);
+
+   EXPECT_STREQ("double", tds.GetTypeName("col1").c_str());
+   EXPECT_STREQ("double", tds.GetTypeName("col2").c_str());
+   EXPECT_STREQ("bool", tds.GetTypeName("col3").c_str());
+   EXPECT_STREQ("double", tds.GetTypeName("col4").c_str());
+   EXPECT_STREQ("Long64_t", tds.GetTypeName("col5").c_str());
+   EXPECT_STREQ("Long64_t", tds.GetTypeName("col6").c_str());
+   EXPECT_STREQ("std::string", tds.GetTypeName("col7").c_str());
+   EXPECT_STREQ("std::string", tds.GetTypeName("col8").c_str());
+
+   // determine type by reading only the first line
+   RCsvDS tds2(fileName4, true, ',', -1LL, "NaN", 0U);
+
+   EXPECT_STREQ("double", tds2.GetTypeName("col1").c_str());
+   EXPECT_STREQ("double", tds2.GetTypeName("col2").c_str());
+   EXPECT_STREQ("bool", tds2.GetTypeName("col3").c_str());
+   EXPECT_STREQ("double", tds2.GetTypeName("col4").c_str());
+   EXPECT_STREQ("Long64_t", tds2.GetTypeName("col5").c_str());
+   EXPECT_STREQ("double", tds2.GetTypeName("col6").c_str());
+   EXPECT_STREQ("std::string", tds2.GetTypeName("col7").c_str());
+   EXPECT_STREQ("double", tds2.GetTypeName("col8").c_str());
+}
+
+TEST(RCsvDS, SetCustomFillValue)
+{
+   ROOT::DisableImplicitMT(); // to allow usage of display
+
+   auto tdf = ROOT::RDF::MakeCsvDataFrame(fileName4);
```

Comment:
rdf? :)

Reply:
I tried to be consistent with the file. I am unsure what `tdf` stands for, but I saw it is used everywhere in the file for `MakeCsvDataFrame`

Reply:
tdf = TDataFrame, former name of RDataFrame :) You can just use the current naming.